pkg/net/coap/service/config.go

```diff
@@ -0,0 +1,144 @@
+package service
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/plgd-dev/hub/v2/pkg/config"
+	certManagerServer "github.com/plgd-dev/hub/v2/pkg/security/certManager/server"
+)
+
+type Protocol string
+
+const (
+	TCP Protocol = "tcp"
+	UDP Protocol = "udp"
+)
+
+// Config represents configuration of coap service
+type Config struct {
+	Addr              string                  `yaml:"address" json:"address"`
+	Protocols         []Protocol              `yaml:"protocols" json:"protocols"`
+	MaxMessageSize    uint32                  `yaml:"maxMessageSize" json:"maxMessageSize"`
+	MessagePoolSize   int                     `yaml:"messagePoolSize" json:"messagePoolSize"`
+	BlockwiseTransfer BlockwiseTransferConfig `yaml:"blockwiseTransfer" json:"blockwiseTransfer"`
+	TLS               TLSConfig               `yaml:"tls" json:"tls"`
+	InactivityMonitor *InactivityMonitor      `yaml:"inactivityMonitor,omitempty" json:"inactivityMonitor,omitempty"`
+	KeepAlive         *KeepAlive              `yaml:"keepAlive,omitempty" json:"keepAlive,omitempty"`
+}
+
+func (c *Config) GetTimeout() time.Duration {
+	if c.KeepAlive != nil {
+		return c.KeepAlive.Timeout
+	}
+	return c.InactivityMonitor.Timeout
+}
+
+func (c *Config) Validate() error {
+	if c.Addr == "" {
+		return fmt.Errorf("address('%v')", c.Addr)
+	}
+	if c.MaxMessageSize <= 64 {
+		return fmt.Errorf("maxMessageSize('%v')", c.MaxMessageSize)
+	}
+	if c.MessagePoolSize < 0 {
+		return fmt.Errorf("messagePoolSize('%v')", c.MessagePoolSize)
+	}
+	if len(c.Protocols) == 0 {
+		return fmt.Errorf("protocols('%v')", c.Protocols)
+	}
+	for i := 0; i < len(c.Protocols); i++ {
+		switch c.Protocols[i] {
+		case TCP, UDP:
+		default:
+			return fmt.Errorf("protocols[%v]('%v')", i, c.Protocols[i])
+		}
+	}
+	if err := c.BlockwiseTransfer.Validate(); err != nil {
+		return fmt.Errorf("blockwiseTransfer.%w", err)
+	}
+	if err := c.TLS.Validate(); err != nil {
+		return fmt.Errorf("tls.%w", err)
+	}
+	if c.InactivityMonitor != nil {
+		if err := c.InactivityMonitor.Validate(); err != nil {
+			return fmt.Errorf("inactivityMonitor.%w", err)
+		}
+	}
+	if c.KeepAlive != nil {
+		if err := c.KeepAlive.Validate(); err != nil {
+			return fmt.Errorf("keepAlive.%w", err)
+		}
+	}
+	if c.KeepAlive == nil && c.InactivityMonitor == nil {
+		return fmt.Errorf("keepAlive or inactivityMonitor must be set")
+	}
+
+	return nil
+}
+
+type TLSConfig struct {
+	Enabled                        *bool                    `yaml:"enabled,omitempty" json:"enabled,omitempty"`
```

Comment:
We need 3 values or why else the pointer instead of value here?

Reply:
I don't show these values when print configuration (hidden) - `omitempty` works only when it is enabled.
For eg: For DPS `Enabled` doesn't have sense in configuration. So in config.yaml and I don't show this attribute.
Similar for `MonitoringInactivity` (DPS) and `KeepAlive` (CoapGW)